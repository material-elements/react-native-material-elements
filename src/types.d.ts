export * from './components/types';
export * from './types/index';
