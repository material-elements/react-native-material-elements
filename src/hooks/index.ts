export * from './useRestyle';
export * from './useThemedProps';
