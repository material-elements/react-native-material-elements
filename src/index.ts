export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
