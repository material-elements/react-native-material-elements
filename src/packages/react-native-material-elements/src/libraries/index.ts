export * from './themes';
export * from './style';
