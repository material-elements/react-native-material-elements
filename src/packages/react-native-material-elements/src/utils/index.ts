export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
