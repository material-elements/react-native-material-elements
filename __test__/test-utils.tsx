import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
