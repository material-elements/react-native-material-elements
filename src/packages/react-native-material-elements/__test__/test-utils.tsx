import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
