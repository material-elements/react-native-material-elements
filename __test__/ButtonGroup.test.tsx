import React from 'react';
import { Button, ButtonGroup } from '../src';
import { render } from './test-utils';

describe('ButtonGroup', () => {
  it('should render correctly with default props', () => {
    const { toJSON } = render(<ButtonGroup />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render child button component correctly', () => {
    const { toJSON } = render(
      <ButtonGroup>
        <Button label="Save" />
        <Button label="Delete" />
      </ButtonGroup>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
