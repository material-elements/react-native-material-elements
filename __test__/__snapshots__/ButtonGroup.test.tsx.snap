// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ButtonGroup should render child button component correctly 1`] = `
<View
  style={
    [
      {},
      {},
      undefined,
    ]
  }
>
  <View
    style={
      [
        {},
        {},
        [
          {},
          [
            {
              "flex": 1,
            },
          ],
        ],
      ]
    }
  >
    <View
      style={
        [
          {},
          {},
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#03A9F4",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": 0,
            "borderColor": "#EEEEEE",
            "borderLeftWidth": 0,
            "borderRadius": 8,
            "borderRightWidth": 0,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": 0,
            "display": "flex",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "overflow": "hidden",
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            collapsable={false}
            style={
              {
                "color": "#FAFAFA",
                "fontSize": 14,
              }
            }
          >
            Save
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {},
        {},
        [
          {},
          [
            {
              "flex": 1,
            },
          ],
        ],
      ]
    }
  >
    <View
      style={
        [
          {},
          {},
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#03A9F4",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": undefined,
            "borderColor": "#EEEEEE",
            "borderLeftWidth": 1,
            "borderRadius": 8,
            "borderRightWidth": 0,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": undefined,
            "display": "flex",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "overflow": "hidden",
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            collapsable={false}
            style={
              {
                "color": "#FAFAFA",
                "fontSize": 14,
              }
            }
          >
            Delete
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`ButtonGroup should render correctly with default props 1`] = `
<View
  style={
    [
      {},
      {},
      undefined,
    ]
  }
/>
`;
